Source/WebCore/svg/animation/SVGSMILElement.cpp

```diff
@@ -1160,6 +1163,9 @@ bool SVGSMILElement::progress(SMILTime elapsed, SVGSMILElement& firstAnimation,
         if (oldActiveState == Inactive)
             startedActiveInterval();
 
+        if (repeat && repeat != m_lastRepeat)
+            smilEventSender().dispatchEventSoon(*this, eventNames().repeatEventEvent);
```

Comment:
If we dispatched one of the `repeat` events here, maybe we should decrement `repeat` by 1 so we don't repeat an extra time?